src/Microsoft.ML.TorchSharp/NasBert/NerTrainer.cs

```diff
@@ -155,6 +157,40 @@ private protected override torch.Tensor CreateTargetsTensor(ref List<TargetType>
                 return torch.tensor(targetArray, device: Device);
             }
 
+            private protected override torch.Tensor PrepareRowTensor(ref VBuffer<uint> target)
+            {
+                ReadOnlyMemory<char> sentenceRom = default;
+                Sentence1Getter(ref sentenceRom);
+                var sentence = sentenceRom.ToString();
+                Tensor t;
+                var encoding = Tokenizer.Encode(sentence);
+
+                if (target.Length != encoding.Tokens.Count)
+                {
+                    var targetIndex = 0;
+                    var targetEditor = VBufferEditor.Create(ref target, encoding.Tokens.Count);
+                    var newValues = targetEditor.Values;
+                    for (var i = 0; i < encoding.Tokens.Count; i++)
+                    {
+                        if (Tokenizer.Model.IsFirstTokenInWord(encoding.Tokens[i]))
```

Comment:
>IsFirstTokenInWord [](http://example.com/codeflow?start=44&length=18)

I think the name of this method is not accurate because you can get a work with starting token not having `Ġ`. 
here is example I found:
```
text = '(Besides that there should be more restaurants like it around the city).'
Token (str): ['<s>', 'Ġ(', 'Besides', 'Ġthat', 'Ġthere', 'Ġshould', 'Ġbe', 'Ġmore', 'Ġrestaurants', 'Ġlike', 'Ġit', 'Ġaround', 'Ġthe', 'Ġcity', ')', 'Ġ.', '</s>']
```

I would instead call this method as TokenStartsWithSpace.

Reply:
Good point. Thats how it was called in the MSR code, but I think your name makes more sense here. I'll get it updated.

Reply:
From that perspective - would @tarekgh's example break our scenario?  What if a named entity is proceeded with a `(` - how does the model identify that?